File: backend/app/schemas/knowledge_base.py
from datetime import datetime
from pydantic import BaseModel

class KnowledgeBaseCreate(BaseModel):
    name: str
    description: str = ""

class KnowledgeBaseUpdate(BaseModel):
    name: str
    description: str = ""

class KnowledgeBaseResponse(BaseModel):
    id: int
    name: str
    description: str = ""
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/knowledgeBase.ts
export interface KnowledgeBase {
  id: number
  name: string
  description: string
  created_at: string
  updated_at: string
}

export interface CreateKnowledgeBaseData {
  name: string
  description: string
}

export interface UpdateKnowledgeBaseData {
  name: string
  description: string
}